pkg/kv/kvserver/replica_raftstorage.go

```diff
@@ -771,6 +771,24 @@ func clearRangeData(
 	return nil
 }
 
+// maybeThrottleSnapshot determines whether throttling should be applied to
+// avoid overloading the store. If so, the method blocks and returns a
+func (s *Store) maybeThrottleSnapshot(ctx context.Context) {
```

Comment:
Shouldn't this be a method on `Replica`?

Reply:
Updated. At the moment the only source of delay will be engine delay. However, to be consistent with AddSST delays, I included both. I've removed engine backpressure delay and included only the total (presuming the total is the metric that is looked at more regularly for AddSST).